src/engine/TransitivePathBase.cpp

```diff
@@ -63,76 +63,87 @@ TransitivePathBase::decideDirection() {
-void TransitivePathBase::fillTableWithHull(IdTable& table, const Map& hull,
-                                           size_t startSideCol,
-                                           size_t targetSideCol) const {
-  CALL_FIXED_SIZE((std::array{table.numColumns()}),
-                  &TransitivePathBase::fillTableWithHullImpl, this, table, hull,
-                  startSideCol, targetSideCol);
+Result::Generator TransitivePathBase::fillTableWithHull(NodeGenerator hull,
+                                                        size_t startSideCol,
+                                                        size_t targetSideCol,
+                                                        bool yieldOnce) const {
+  return ad_utility::callFixedSize(getResultWidth(), [&]<size_t WIDTH>() {
+    return fillTableWithHullImpl<0, WIDTH>(std::move(hull), startSideCol,
+                                           targetSideCol, yieldOnce);
+  });
 }
 
 // _____________________________________________________________________________
-template <size_t WIDTH>
-void TransitivePathBase::fillTableWithHullImpl(IdTable& tableDyn,
-                                               const Map& hull,
-                                               size_t startSideCol,
-                                               size_t targetSideCol) const {
-  IdTableStatic<WIDTH> table = std::move(tableDyn).toStatic<WIDTH>();
-  size_t rowIndex = 0;
-  for (auto const& [node, linkedNodes] : hull) {
+template <size_t INPUT_WIDTH, size_t OUTPUT_WIDTH>
+Result::Generator TransitivePathBase::fillTableWithHullImpl(
+    NodeGenerator hull, size_t startSideCol, size_t targetSideCol,
+    bool yieldOnce, size_t skipCol) const {
+  ad_utility::Timer timer{ad_utility::Timer::Stopped};
+  size_t outputRow = 0;
+  size_t inputRow = 0;
+  IdTableStatic<OUTPUT_WIDTH> table{getResultWidth(), allocator()};
+  std::vector<LocalVocab> storedLocalVocabs;
+  for (auto& [node, linkedNodes, localVocab, idTable] : hull) {
+    timer.cont();
+    if (!yieldOnce) {
+      table.reserve(linkedNodes.size());
+    }
+    std::optional<IdTableView<INPUT_WIDTH>> inputView = std::nullopt;
+    if (idTable != nullptr) {
+      inputView = idTable->template asStaticView<INPUT_WIDTH>();
+    }
     for (Id linkedNode : linkedNodes) {
       table.emplace_back();
-      table(rowIndex, startSideCol) = node;
-      table(rowIndex, targetSideCol) = linkedNode;
+      table(outputRow, startSideCol) = node;
+      table(outputRow, targetSideCol) = linkedNode;
 
-      rowIndex++;
+      if (inputView.has_value()) {
+        copyColumns<INPUT_WIDTH, OUTPUT_WIDTH>(inputView.value(), table,
+                                               inputRow, outputRow, skipCol);
+      }
+
+      outputRow++;
+    }
+    inputRow++;
+    // Reset row to zero so the next `IdTable` is accessed correctly. The
+    // implicit assumption here is that every row in all `IdTable`s corresponds
+    // to exactly one start node.
```

Comment:
I just realized the index calculation is off when there are starting nodes without any matches. So just incrementing `inputRow` by one each time is wrong and needs to be fixed by passing the index to here, or by yielding empty entries as well.